app/components/markdown/markdown.js

```diff
@@ -169,6 +179,41 @@ export default class Markdown extends PureComponent {
         // Construct the text style based off of the parents of this node since RN's inheritance is limited
         const style = this.computeTextStyle(this.props.baseTextStyle, context);
 
+        if (this.props.enableLatex && this.props.enableInlineLatex) {
+            // eslint-disable-next-line no-useless-escape
+            const inlineLatexRegEx = /\$([^\$\n]+)\$(?!\w)/;
+
+            //Search for match
+            const match = inlineLatexRegEx.exec(literal);
+
+            if (match && !(/\w/).exec(literal[match.index - 1])) { //Extra check to see if there is no word character in front of the matched latex code
+                const firstPart = literal.slice(0, match.index);
+                let latexCode = match[1];
+                const lastPart = literal.slice(match.index + match[0].length);
+
+                latexCode = unEscapeLatexCodeInText(latexCode); //Unescape code
+
+                latexCode = latexCode.replace('&', '\\&'); //For some reason the slash before an & character is missing when you type it.
+
+                return (
+                    <Text
+                        style={{lineHeight: this.state.inlineLatexHeight}}
+                    >
+                        {this.renderText({context, literal: firstPart})}
```

Comment:
this is calling itself, is that expected?

Also, perhaps you can extract all this if block to it's own function for readability.

Reply:
That behaviour is expected. What this function does is check if there is an inline latex match. If there is a match it splits the text up in text and latex code. It calls itself on the text part to check if there are maybe more inline latex code blocks and renders the latex code.

I'll split this function up.